### DISCUSSION TITLE
Using E1DET pin in BTT SKR 2.0 as a limit pin.

### DISCUSSION THREAD

**Discussion by krishchan007** [grblHAL/STM32F4xx#98] [Q&A] [answered]
`// The normal limit pin for E1 is PCA0, but bit 0 already has an interrupt (Z_LIMIT_PIN).
// PC15 is normally used for PWRDET but is used for M4_LIMIT_PIN instead.
// If using TMC drivers, jumper from PWRDET connector pin 3 to DIAG pin on driver.
`
Ignoring these lines I used PCA0 as limit for Axis B. Both the Z-LIMIT and B-LIMIT was working without any issues. What am i missing here?

**Comment by terjeio** [comment, marked as answer]
>  Both the Z-LIMIT and B-LIMIT was working without any issues.

When hard limits is enabled as well?

>  What am i missing here?

A given pin number regardless of the port it belongs to can only be assigned to one interrupt line. E.g. A0 and C0 cannot both trigger an interrupt, the last assigned will "win". This is a limitation of the STM32 architecture - see section 12.2.5 in the reference manual.

**Reply by krishchan007** [reply]
If hard limits are not enabled, will there be any problem? I mean as soft limits, can it affect the homing sequence. Just wondering what will happen.

**Reply by terjeio** [reply]
> If hard limits are not enabled, will there be any problem?

Short answer: no.
